presto-main/src/main/java/com/facebook/presto/operator/NativeExecutionOperator.java

```diff
@@ -0,0 +1,247 @@
+        this.systemMemoryContext = operatorContext.localSystemMemoryContext();
+        this.taskSourceCodec = requireNonNull(taskSourceCodec, "taskSourceJsonCodec is null");
+        this.planFragmentCodec = requireNonNull(planFragmentCodec, "planFragmentCodec is null");
+        this.tableWriteInfoCodec = requireNonNull(tableWriteInfoCodec, "tableWriteInfoCodec is null");
+        this.planFragment = requireNonNull(planFragment, "planFragment is null");
+        this.tableWriteInfo = requireNonNull(tableWriteInfo, "tableWriteInfo is null");
+        this.taskSource = null;
+        this.finished = false;
+        this.serde = serde;
+    }
+
+    @Override
+    public OperatorContext getOperatorContext()
+    {
+        return operatorContext;
+    }
+
+    @Override
+    public boolean needsInput()
+    {
+        return false;
+    }
+
+    @Override
+    public void addInput(Page page)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    /*
+     * TODO: The getOutput method will NativeExecutionTask APIs to:
+     *  1. submit the plan to the external process
+     *  2. wait on the getResult method from NativeExecutionTask until it returns (to get the task result from external process)
+     *  3. call getTaskInfo method to get the TaskInfo and propagate it
+     *  4. Deserialize the serialized page to presto Page and return it back
+     */
+    @Override
+    public Page getOutput()
+    {
+        if (finished) {
+            return null;
+        }
+        finished = true;
+        return null;
+    }
+
+    @Override
+    public void finish()
+    {
+        finished = true;
+    }
+
+    @Override
+    public boolean isFinished()
+    {
+        return finished;
+    }
+
+    @Override
+    public PlanNodeId getSourceId()
+    {
+        return sourceId;
+    }
+
+    @Override
+    public Supplier<Optional<UpdatablePageSource>> addSplit(ScheduledSplit split)
+    {
+        requireNonNull(split, "split is null");
+        checkState(this.taskSource == null, "NativeEngine operator split already set");
+
+        if (finished) {
+            return Optional::empty;
+        }
+
+        this.taskSource = new TaskSource(split.getPlanNodeId(), ImmutableSet.of(split), true);
+
+        Object splitInfo = split.getSplit().getInfo();
+        Map<String, String> infoMap = split.getSplit().getInfoMap();
+
+        //Make the implicit assumption that if infoMap is populated we can use that instead of the raw object.
+        if (infoMap != null && !infoMap.isEmpty()) {
+            operatorContext.setInfoSupplier(Suppliers.ofInstance(new SplitOperatorInfo(infoMap)));
+        }
+        else if (splitInfo != null) {
+            operatorContext.setInfoSupplier(Suppliers.ofInstance(new SplitOperatorInfo(splitInfo)));
+        }
+
+        return Optional::empty;
+    }
+
+    @Override
+    public void noMoreSplits()
+    {
+        if (taskSource == null) {
+            finished = true;
+        }
```

Comment:
I'm trying to understand, when noMoreSplits() is called, it means the driver(Presto driver) tells this operator that there will be no more splits going in. taskSource being null means we have not added any splits. Does this mean we are not going to process anything and call it done?